src/Receipt/DetailGroup.php

```diff
@@ -0,0 +1,116 @@
+<?php
+
+namespace Give\Receipt;
+
+/**
+ * Class DetailGroup
+ *
+ * This class represent receipt detail group as object and you can add ass many as you want detail item.
+ *
+ * @since 2.7.0
+ * @package Give\Receipt
+ */
+abstract class DetailGroup {
+	/**
+	 * Group heading.
+	 *
+	 * @since 2.7.0
+	 * @var string $heading
+	 */
+	public $heading = '';
+
+	/**
+	 * Donation Id.
+	 *
+	 * since 2.7.0
+	 *
+	 * @var int $donationId
+	 */
+	public $donationId;
+
+	/**
+	 * Array of detail item objects.
+	 *
+	 * @since 2.7.0
+	 * @var Detail[]
+	 */
+	protected $details = [];
+
+	/**
+	 * Array of detail item class names.
+	 *
+	 * @since 2.7.0
+	 * @var Detail[]
+	 */
+	protected $detailsList;
+
+	/**
+	 * DetailGroup constructor.
+	 *
+	 * @since 2.7.0
+	 * @param int $donationId
+	 */
+	public function __construct( $donationId ) {
+		$this->donationId = $donationId;
+
+		$this->setupDetailsGroup();
+	}
+
+	/**
+	 * Setup receipt.
+	 *
+	 * @since 2.7.0
+	 */
+	private function setupDetailsGroup() {
+		/**
+		 * Filter the detail item class names.
+		 *
+		 * Developer can use this filter hook to register there detail item.
+		 */
+		$this->detailsList = apply_filters( 'give_receipt_register_detail_item', $this->detailsList, $this );
+	}
+
+	/**
+	 * Get detail item object.
+	 *
+	 * @since 2.7.0
+	 * @param string $class
+	 * @return Detail
+	 */
+	public function getDetailItemObject( $class ) {
+		if ( ! in_array( $class, $this->details, true ) ) {
+			$classNames              = $this->getDetailsList();
+			$detailGroupClassName    = $classNames[ array_search( $class, $classNames, true ) ];
+			$this->details[ $class ] = null;
+
+			if ( $detailGroupClassName ) {
+				$this->details[ $class ] = new $detailGroupClassName( $this->donationId );
+			}
+		}
+
+		return $this->details[ $class ];
+	}
+
+	/**
+	 * Get detail item class names.
+	 *
+	 * @since 2.7.0
+	 * @return array
+	 */
+	public function getDetailsList() {
+		return $this->detailsList;
+	}
+
+	/**
+	 * Return whether or not to render details group.
+	 *
+	 * Each detail item need specific context/condition to render. For example recurring related detail will on render if donation is a subscription.
+	 * We can use this function in receipt of form template to prevent unnecessary detail group output.
+	 *
+	 * @return bool
+	 * @since 2.7.0
+	 */
+	public function canShow() {
```

Comment:
@ravinderk So this will be more useful later when more complexity is added to determine this?

Reply:
@JasonTheAdams yes. For example, I am using this in recurring addon-on for Subscription detail. Few details need context to show and that function will play an important role where we return bool value. to toggle a detail group.

ref: https://github.com/impress-org/give-recurring/pull/942/files#diff-e456b609257e9e1171f82aa5533985ceR89